File: examples/objective-c-demo/objective-c-demo/PlayerViewController.m
#import "PlayerViewController.h"
#import "HomeNavViewController.h"

@interface PlayerViewController (){
    AVPlayer *_avplayer;
    AVPlayerViewController *_avplayerController;
    MLYAVPlayerPlugin *_plugin;
}
@end

@implementation PlayerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
    
    _plugin = [[MLYAVPlayerPlugin alloc] init];
    NSError *error = nil;
    
    [MLYDriver initializeAndReturnError:  &error :^(MLYDriverOptions * option) {
        [[option client] setId: clientID];
        
        [_plugin adapt:_avplayerController];
    }];
    
    if(error != nil){
        NSLog(@"MLYDriver initialize err:%@",error.description);
    }
    NSURL *url = [NSURL URLWithString: videoURL];
    AVPlayerItem *item = [AVPlayerItem playerItemWithURL:url];
    [_avplayer replaceCurrentItemWithPlayerItem:item];
    [_avplayer play];
}

- (void)setupUI {
    _avplayer = [[AVPlayer alloc] init];
    _avplayerController = [AVPlayerViewController new];
    _avplayerController.showsPlaybackControls = true;
    _avplayerController.view.backgroundColor = UIColor.darkGrayColor;
    _avplayerController.player = _avplayer;
    
    [self addChildViewController:_avplayerController];
    [[self view] addSubview:_avplayerController.view];
}

- (void)viewDidDisappear:(BOOL)animated{
    [_plugin deactivate];
    [_avplayer replaceCurrentItemWithPlayerItem:nil];
    [super viewDidDisappear: animated];
}
  
@end

File: Examples/swift/swift/PlayerViewController.swift
import AVFoundation
import AVKit
import MLYSDK
import UIKit

class PlayerViewController: UIViewController {
    lazy var player:AVPlayer? = AVPlayer()
    lazy var playerViewController = {
        let controller = AVPlayerViewController()
        controller.showsPlaybackControls = true
        controller.view.translatesAutoresizingMaskIntoConstraints = false
        controller.view.backgroundColor = .darkGray
        controller.player = self.player
        return controller
    }()

    lazy var plugin = MLYAVPlayerPlugin()

    override func viewDidLoad() {
        super.viewDidLoad()
        self.setupUI()
        do {
            try MLYDriver.initialize { options in
                options.client.id = clientID
            }
            self.plugin.adapt(self.playerViewController)
        } catch {
            print(error)
        }
        guard let player = self.player else {
            return
        }
        let url = URL(string:videoURL)!
        let playerItem = AVPlayerItem(url: url)
        player.replaceCurrentItem(with: playerItem)
        player.play()
    }

    func setupUI() {
        self.view.backgroundColor = .black
        self.addChild(self.playerViewController)
        self.view.addSubview(self.playerViewController.view)
        self.playerViewController.view.frame = .init(x: 0, y: 0, width: UIView.screenWidth(), height: UIView.screenHeight())
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        self.plugin.deactivate()
        self.player?.replaceCurrentItem(with: nil)
        self.player = nil
        super.viewDidDisappear(animated)
    }
}
